src/charm.py

```diff
@@ -202,85 +225,102 @@ def reconfigure_nats(self):
+                    "/usr/lib/nagios/plugins/check_tcp",
+                    "-H",
+                    str(self.client.listen_address),
+                    "-p",
+                    str(self.model.config["client-port"]),
+                ],
+                name=check_name,
+            )
             self.nrpe_client.commit()
 
-        tenv = Environment(loader=FileSystemLoader('templates'))
-        template = tenv.get_template('nats.cfg.j2')
+        tenv = Environment(loader=FileSystemLoader("templates"))
+        template = tenv.get_template("nats.cfg.j2")
         rendered_content = template.render(ctxt)
-        content_hash = self.generate_content_hash(rendered_content)
+        content_hash = self._generate_content_hash(rendered_content)
         old_hash = self.state.nats_config_hash
         if old_hash != content_hash:
-            logging.info(f'Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}')
+            logging.info(
+                f"Config has changed - re-rendering a template to {self.NATS_SERVER_CONFIG_PATH}"
+            )
             self.state.nats_config_hash = content_hash
             self.NATS_SERVER_CONFIG_PATH.write_text(rendered_content)
             if self.state.is_started:
-                subprocess.check_call(['systemctl', 'restart', self.NATS_SERVICE])
+                subprocess.check_call(["systemctl", "restart", self.NATS_SERVICE])
         self.client.expose_nats(auth_token=self.state.auth_token)
 
-        client_port = self.model.config['client-port']
-        if (client_port is None or client_port == 0) and (self.state.client_port is not None or len(self.state.client_port) > 0):
+        client_port = self.model.config["client-port"]
+        if (client_port is None or client_port == 0) and (
+            self.state.client_port is not None or len(self.state.client_port) > 0
+        ):
             self._close_port(self.state.client_port)
         else:
-            port = '{}/tcp'.format(client_port)
+            port = "{}/tcp".format(client_port)
             if self.state.client_port is not None and port != self.state.client_port:
                 self._close_port(self.state.client_port)
             self._open_port(port)
             self.state.client_port = port
 
         self.unit.status = ActiveStatus()
 
-    def get_auth_token(self, length=None):
-        '''Generate a random auth token.'''
+    @classmethod
+    def get_auth_token(cls, length=None):
+        """Generate a random auth token."""
         if not isinstance(length, int):
-            raise RuntimeError('invalid length provided for a token')
+            raise RuntimeError("invalid length provided for a token")
         alphanumeric_chars = string.ascii_letters + string.digits
         rng = random.SystemRandom()
-        return ''.join([rng.choice(alphanumeric_chars) for _ in range(length)])
+        return "".join([rng.choice(alphanumeric_chars) for _ in range(length)])
```

Comment:
I'd recommend using the python [`secrets`](https://docs.python.org/3/library/secrets.html) module for this, and possibly storing the result in a Juju Secret? (Juju 3.0+)